build-system/server/app-index/util/listing.js

```diff
@@ -35,8 +34,8 @@ async function getListing(rootPath, basepath) {
   }
 
   try {
-    if ((await fs.statAsync(path)).isDirectory()) {
-      return fs.readdirAsync(path);
+    if (fs.statSync(path).isDirectory()) {
+      return fs.promises.readdir(path);
```

Comment:
So this is an `async` function that returns a `Promise`? Making its return type `Promise<Promise<string>>`?

Reply:
I recognize it's out of scope for this PR, but that's weird :eyes:

Reply:
Async functions only wrap non-promise return values.

Reply:
Returning a promise from `async` folds the promise into the chain.

Reply:
Oooh interesting, I didn't know that!